bridge-adapters/kinetic-bridgehub-adapter-cherwell/src/main/java/com/kineticdata/bridgehub/adapter/cherwell/CherwellApiHelper.java

```diff
@@ -0,0 +1,232 @@
+            if (responseCode >= 400) {
+                handleFailedRequest(responseCode);
+            }
+        }
+        catch (IOException e) {
+            throw new BridgeError(
+                "Unable to make a connection to the Harvest service server.", e);
+        }
+        
+        return output;
+    }
+
+    public JSONObject executePostRequest (String path, String body) throws BridgeError{
+
+        String url = baseUrl + path;
+        JSONObject output;
+        // System time used to measure the request/response time
+        long start = System.currentTimeMillis();
+
+        try (
+                CloseableHttpClient client = HttpClients.createDefault()
+        ) {
+            HttpResponse response;
+            HttpPost post = new HttpPost(url);
+
+            // Append HTTP BASIC Authorization header to HttpGet call
+            post.setHeader("Content-Type", "application/json");
+            post.setHeader("Accept", "application/json");
+            post.setHeader("Authorization", "Bearer " + token);
+
+            String payload = body;
+            StringEntity stringEntity = new StringEntity(payload);
+            post.setEntity(stringEntity);
+
+            response = client.execute(post);
+            LOGGER.debug("Recieved response from \"{}\" in {}ms.",
+                    url,
+                    System.currentTimeMillis()-start);
+
+            int responseCode = response.getStatusLine().getStatusCode();
+            LOGGER.trace("Request response code: " + responseCode);
+
+            HttpEntity entity = response.getEntity();
+
+            // Confirm that response is a JSON object
+            output = parseResponse(EntityUtils.toString(entity));
+
+            // Handle all other failed repsonses
+            if (responseCode >= 400) {
+                handleFailedRequest(responseCode);
+            }
+        }
+        catch (IOException e) {
+            throw new BridgeError(
+                    "Unable to make a connection to the Harvest service server.", e);
+        }
+
+        return output;
+    }
+    
+    // Get a JWT to be used with subsequent requests.
+    public void getToken () throws BridgeError {
+        String url = baseUrl + "/token";
+        
+        try (
+            CloseableHttpClient client = HttpClients.createDefault()
+        ) {
+            HttpResponse response;
+            HttpPost httpPost = new HttpPost(url);
+      
+            // Create entity with username and pass for use in the Post.
+            List<NameValuePair> form = new ArrayList<>();
+            form.add(new BasicNameValuePair("username", username));
+            form.add(new BasicNameValuePair("password", password));
+            form.add(new BasicNameValuePair("client_id", clientId));
+            form.add(new BasicNameValuePair("grant_type", "password"));
+            UrlEncodedFormEntity requestEntity = new UrlEncodedFormEntity(form, Consts.UTF_8);
```

Comment:
Prefer to use `StandardCharsets.UTF_8` instead of `Consts.UTF_8`. But maybe in this case it doesn't matter since Both the UrlEncodedFormEntity class and Consts class are part of the Apache HTTP Client library. But usually like to use Java standard library classes if they are available.